sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -206,21 +242,42 @@ Scheduler::GraphBuilder::insertMemCpyCmd(MemObjRecord *Record, Requirement *Req,
           !"Inappropriate alloca command. AllocaSubBufCommand was expected.");
   }
 
-  // Full copy of buffer is needed to avoid loss of data that may be caused
-  // by copying specific range form host to device and backwards.
-  MemCpyCommand *MemCpyCmd = new MemCpyCommand(
-      *AllocaCmdSrc->getRequirement(), AllocaCmdSrc,
-      *AllocaCmdDst->getRequirement(), AllocaCmdDst, AllocaCmdSrc->getQueue(),
-      AllocaCmdDst->getQueue(), UseExclusiveQueue);
+  const QueueImplPtr &SrcQueue = AllocaCmdSrc->getQueue();
+  Command *NewCmd = nullptr;
+
+  if (AllocaCmdSrc->MLinkedAllocaCmd == AllocaCmdDst) {
+    NewCmd = insertMapUnmapForLinkedCmds(AllocaCmdSrc, AllocaCmdDst);
+  } else {
+
+    if (AllocaCmdDst->MLinkedAllocaCmd && !AllocaCmdDst->MIsActive) {
+      NewCmd = insertMapUnmapForLinkedCmds(AllocaCmdSrc, AllocaCmdDst);
```

Comment:
Capturing offline discussion: this looks like a bug, should probably be:
`NewCmd = insertMapUnmapForLinkedCmds(AllocaCmdDst->MLinkedAllocaCmd, AllocaCmdDst);`

Reply:
It turns out that the code is currently unreachable. Removing.